src/Controller/AdminController.php

```diff
@@ -777,14 +783,47 @@ public function deliveriesAction(Request $request,
             return $response;
         }
 
-        $sections = $this->getDoctrine()
-            ->getRepository(Delivery::class)
-            ->getSections();
+        $filters = [
+            'query' => null,
+            'range' => null,
+        ];
+        if (!is_null($request->query->get('q'))) {
+
+            $filters['query'] = $request->query->get('q');
+
+            $locale = $request->getLocale();
+            $search = new \Psonic\Search($client);
+            $search->connect($this->getParameter('sonic_secret_password'));
+
+            $ids = $search->query('store:*:deliveries', $this->getParameter('sonic_namespace'),
+                $request->query->get('q'), $limit = null, $offset = null, Languages::getAlpha3Code($locale));
+
+            $search->disconnect();
+
+            $ids = array_filter($ids);
+
+            $qb = $this->getDoctrine()->getRepository(Delivery::class)->findByIds($ids);
 
-        if ($request->query->has('section')) {
-            $qb = $sections[$request->query->get('section')];
         } else {
-            $qb = $sections['today'];
+            $sections = $this->getDoctrine()
+            ->getRepository(Delivery::class)->getSections();
+
+
+            if ($request->query->has('section')) {
+                $qb = $sections[$request->query->get('section')];
+            } else {
+                $qb = $sections['today'];
+            }
+        }
+
+        if (!is_null($request->query->get('start_at')) && !is_null($request->query->get('end_at'))) {
```

Comment:
You can use the `has()` method

```suggestion
        if ($request->query->has('start_at') && $request->query->has('end_at')) {
```